pkg/apis/events/v1alpha1/channel_types.go

```diff
@@ -60,25 +59,21 @@ type ChannelSpec struct {
 	// Topic and Subscriptions. The value of the secret entry must be a
 	// service account key in the JSON format
 	// (see https://cloud.google.com/iam/docs/creating-managing-service-account-keys).
+	// +optional
 	Secret *corev1.SecretKeySelector `json:"secret,omitempty"`
 
 	// Project is the ID of the Google Cloud Project that the Pub/Sub
 	// Topic and Subscriptions will be created in.
```

Comment:
And here, esp. the bit about metadata?

Reply:
I want to do this as a followup because it is a concept that touches all the resources. I will add an issue to track.